drivers: esp32: esp_wifi_drv strncpy warning

<!--
**Notes**
Github Discussions (https://github.com/zephyrproject-rtos/zephyr/discussions)
are available to first verify that the issue is a genuine Zephyr bug and not a
consequence of Zephyr services misuse.

This issue list is only for bugs in the main Zephyr code base
(https://github.com/zephyrproject-rtos/zephyr/). If the bug is for a project
fork (such as NCS) specific feature, please open an issue in the fork project
instead.
-->

**Describe the bug**
<!--
A clear and concise description of what the bug is.

Please also mention any information which could help others to understand
the problem you're facing:
 - What target platform are you using?
 - What have you tried to diagnose or workaround this issue?
 - Is this a regression? If yes, have you been able to "git bisect" it to a
   specific commit?
 - ...
-->
When compiling `esp_wifi_drv.c` with `CONFIG_SPEED_OPTIMIZATIONS=y` the compiler creates a warning:
```bash
   esp_wifi_drv.c:736:9: warning: 'strncpy' output may be truncated copying 32 bytes
       from a string of length 32 [-Wstringop-truncation]
      736 |         strncpy(status->ssid, data->status.ssid, WIFI_SSID_MAX_LEN);
```
**To Reproduce**
<!--
Steps to reproduce the behavior:
1. mkdir build; cd build
2. cmake -DBOARD=board\_xyz
3. make
4. See error
-->
```bash
cd tests/boards/espressif/wifi
west build -p -b esp32s3_devkitm/esp32s3/procpu -- -DCONFIG_SPEED_OPTIMIZATIONS=y
```
**Environment (please complete the following information):**
main

--- a/drivers/wifi/esp32/src/esp_wifi_drv.c
+++ b/drivers/wifi/esp32/src/esp_wifi_drv.c
@@ -735,6 +735,7 @@ static int esp32_wifi_status(const struct device *dev, struct wifi_iface_status 
 
 	strncpy(status->ssid, data->status.ssid, WIFI_SSID_MAX_LEN);
 	status->ssid_len = strnlen(data->status.ssid, WIFI_SSID_MAX_LEN);
+	status->ssid[status->ssid_len] = '\0';
 	status->band = WIFI_FREQ_BAND_2_4_GHZ;
 	status->link_mode = WIFI_LINK_MODE_UNKNOWN;
 	status->mfp = WIFI_MFP_DISABLE;

--- a/include/zephyr/net/wifi_mgmt.h
+++ b/include/zephyr/net/wifi_mgmt.h
@@ -419,7 +419,7 @@ struct wifi_scan_params {
  */
 struct wifi_scan_result {
 	/** SSID */
-	uint8_t ssid[WIFI_SSID_MAX_LEN];
+	uint8_t ssid[WIFI_SSID_MAX_LEN + 1];
 	/** SSID length */
 	uint8_t ssid_length;
 	/** Frequency band */
@@ -576,7 +576,7 @@ struct wifi_iface_status {
 	/** SSID length */
 	unsigned int ssid_len;
 	/** SSID */
-	char ssid[WIFI_SSID_MAX_LEN];
+	char ssid[WIFI_SSID_MAX_LEN + 1];
 	/** BSSID */
 	char bssid[WIFI_MAC_ADDR_LEN];
 	/** Frequency band */

